### ISSUE TITLE
🐞Wrong ReactJS version to debug web parts in SharePoint 2019 workbench

### ISSUE THREAD

**Issue by StfBauer** [SharePoint/sp-dev-docs#3145]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea
- releated issue #3128 

#### Expected or Desired Behavior
When the SharePoint 2019 option will be selected it should be possible to run a web part on the SharePoint 2019 server workbench. It seems like the workbench not yet support React16 but a new project comes with React16.

#### Observed Behavior
When a project gets created using the following options.

<img width="959" alt="screenshot 2018-12-19 at 12 30 50" src="https://user-images.githubusercontent.com/5503835/50217766-f528b780-0389-11e9-8f94-e83be4cf23f9.png">

The package.json references React16 instead of React15.

```js
  "dependencies": {
    "react": "16.3.2",
    "react-dom": "16.3.2",
    "@types/react": "16.4.2",
    "@types/react-dom": "16.0.5",
```

Therefor debugging and running the web part in the local workbench runs perfectly but not on a SharePoint 2019 Server.

It leads to the following issue:

<img width="944" alt="screenshot 2018-12-19 at 12 34 39" src="https://user-images.githubusercontent.com/5503835/50217915-797b3a80-038a-11e9-8581-5bc149745610.png">

The solution to this is to downgrade the React components in the `package.json` to use React15.

```js
"dependencies": {
    "react": "15.4.2",
    "react-dom": "15.4.2",
    "@types/react": "0.14.46",
    "@types/react-dom": "0.14.18",
```

After this modification in the `package.json` and performing an `npm update`, the workbench is able to load the web part without any problem.

<img width="929" alt="screenshot 2018-12-19 at 12 39 12" src="https://user-images.githubusercontent.com/5503835/50218131-1ccc4f80-038b-11e9-92ba-5315e0c611a1.png">

So my guess is, and this is most likely the case that the older on-premise workbench is unable to handle ReactJS16 yet. SPO workbench doesn't have this issue but is a newer version.

#### Steps to Reproduce
1. Create a new web part project using "SharePoint 2019 onwards, including SharePoint Online"
1. run `gulp serve`
1. try to add web part to SharePoint 2019 workbench
1. downgrade current ReactJS version to React15
1. ```npm update```
1. Try to add web part again to SharePoint 2019 workbench.

**Comment by kemesfun** [comment]
@StfBauer I am getting React15 in package.json when I repro your steps, please verify that you get React15.

**Comment by StfBauer** [comment]
@kemesfun - Using the latest version on a more than one year old issue no surprise 😃

I did a quick checkup when it was fixed and batch created project from version 1.7.0 to more recent ones. Seems like it was fixed in version 1.8.1.

**SPFX 1.7.0**
```json
{
  "name": "tmp",
  "version": "0.0.1",
  "private": true,
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "build": "gulp bundle",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "react": "16.3.2",
    "react-dom": "16.3.2",
    "@types/react": "16.4.2",
    "@types/react-dom": "16.0.5",
    "@microsoft/sp-core-library": "~1.4.0",
    "@microsoft/sp-webpart-base": "~1.4.0",
    "@microsoft/sp-lodash-subset": "~1.4.0",
    "@microsoft/sp-office-ui-fabric-core": "~1.4.0",
    "@types/webpack-env": "1.13.1",
    "@types/es6-promise": "0.0.33"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "~1.4.1",
    "@microsoft/sp-module-interfaces": "~1.4.1",
    "@microsoft/sp-webpart-workbench": "~1.4.1",
    "gulp": "~3.9.1",
    "@types/chai": "3.4.34",
    "@types/mocha": "2.2.38",
    "ajv": "~5.2.2"
  }
}
```

**SPFx 1.7.1**
```json
{
  "name": "tmp",
  "version": "0.0.1",
  "private": true,
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "build": "gulp bundle",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "react": "16.3.2",
    "react-dom": "16.3.2",
    "@types/react": "16.4.2",
    "@types/react-dom": "16.0.5",
    "@microsoft/sp-core-library": "1.7.1",
    "@microsoft/sp-webpart-base": "1.7.1",
    "@microsoft/sp-lodash-subset": "1.7.1",
    "@microsoft/sp-office-ui-fabric-core": "1.7.1",
    "@types/webpack-env": "1.13.1",
    "@types/es6-promise": "0.0.33"
  },
  "resolutions": {
    "@types/react": "16.4.2"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "1.7.1",
    "@microsoft/sp-tslint-rules": "1.7.1",
    "@microsoft/sp-module-interfaces": "1.7.1",
    "@microsoft/sp-webpart-workbench": "1.7.1",
    "gulp": "~3.9.1",
    "@types/chai": "3.4.34",
    "@types/mocha": "2.2.38",
    "ajv": "~5.2.2"
  }
}
```

**SPFX 1.8.0**

```json
{
  "name": "tmp",
  "version": "0.0.1",
  "private": true,
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "build": "gulp bundle",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "react": "16.7.0",
    "react-dom": "16.7.0",
    "@types/react": "16.4.2",
    "@types/react-dom": "16.0.5",
    "@microsoft/sp-core-library": "~1.4.0",
    "@microsoft/sp-webpart-base": "~1.4.0",
    "@microsoft/sp-lodash-subset": "~1.4.0",
    "@microsoft/sp-office-ui-fabric-core": "~1.4.0",
    "@types/webpack-env": "1.13.1",
    "@types/es6-promise": "0.0.33"
  },
  "resolutions": {
    "@types/react": "16.4.2"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "~1.4.1",
    "@microsoft/sp-module-interfaces": "~1.4.1",
    "@microsoft/sp-webpart-workbench": "~1.4.1",
    "gulp": "~3.9.1",
    "@types/chai": "3.4.34",
    "@types/mocha": "2.2.38",
    "ajv": "~5.2.2"
  }
}
```

**SPFx 1.8.1** -- FIXED

```json
{
  "name": "tmp",
  "version": "0.0.1",
  "private": true,
  "engines": {
    "node": ">=0.10.0"
  },
  "scripts": {
    "build": "gulp bundle",
    "clean": "gulp clean",
    "test": "gulp test"
  },
  "dependencies": {
    "react": "15.6.2",
    "react-dom": "15.6.2",
    "@types/react": "15.6.6",
    "@types/react-dom": "15.5.6",
    "@microsoft/sp-core-library": "~1.4.0",
    "@microsoft/sp-webpart-base": "~1.4.0",
    "@microsoft/sp-lodash-subset": "~1.4.0",
    "@microsoft/sp-office-ui-fabric-core": "~1.4.0",
    "@types/webpack-env": "1.13.1",
    "@types/es6-promise": "0.0.33"
  },
  "resolutions": {
    "@types/react": "15.6.6"
  },
  "devDependencies": {
    "@microsoft/sp-build-web": "~1.4.1",
    "@microsoft/sp-module-interfaces": "~1.4.1",
    "@microsoft/sp-webpart-workbench": "~1.4.1",
    "gulp": "~3.9.1",
    "@types/chai": "3.4.34",
    "@types/mocha": "2.2.38",
    "ajv": "~5.2.2"
  }
}
